### PR TITLE
Feature/4699 task pass risklevelper date to view

### PR DESCRIPTION
**Pull request by KaiTeuber**
## Description
No longer store RiskLevel per Date in the contact journal, use latest risk calculation only
Journal overview will update if the RiskLevel changes while present
Added new UITests
Fixed some old teste to work with the new 15 days limit

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4699

## Screenshots
Simulator Screen Shot - Clone 1 of iPhone 12 Pro - 2021-01-21 at 12 28 34
Simulator Screen Shot - Clone 1 of iPhone 12 Pro - 2021-01-21 at 12 28 56

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/DiaryOverviewViewModel.swift [hunk 1]
```diff
@@ -9,12 +9,36 @@ class DiaryOverviewViewModel {
...
 		}.store(in: &subscriptions)
+
+		homeState?.$riskState
+			.receive(on: RunLoop.main.ocombine)
+			.sink { [weak self] updatedRiskState in
+				switch updatedRiskState {
+				case .risk:
+					self?.refreshTableView?()
+				default:
+					Log.debug("risk state ")
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/DiaryOverviewViewModel.swift:38]: I think this message could be more meaningful, if we want to log this 😅

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/DiaryOverviewViewModel.swift [hunk 2]
```diff
@@ -24,12 +48,22 @@ class DiaryOverviewViewModel {
...
 
 	var numberOfSections: Int {
 		Section.allCases.count
 	}
 
+	var refreshTableView: (() -> Void)?
+
+	var count: Int {
+		days.count
+	}
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/DiaryOverviewViewModel.swift:61]: This seems not to be used

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/__tests__/ContactDiaryStoreTests.swift [hunk 3]
```diff
@@ -434,23 +434,23 @@ class ContactDiaryStoreTests: XCTestCase {
...
 
-	func test_When_cleanupIsCalled_Then_EntriesOlderThen16DaysAreDeleted() {
+	func test_When_cleanupIsCalled_Then_EntriesOlderThen17DaysAreDeleted() {
 		let databaseQueue = makeDatabaseQueue()
 		let store = makeContactDiaryStore(with: databaseQueue)
 
 		let today = Date()
 
-		guard let seventeenDaysAgo = Calendar.current.date(byAdding: .day, value: -17, to: today) else {
+		guard let seventeenDaysAgo = Calendar.current.date(byAdding: .day, value: -18, to: today) else {
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/__tests__/ContactDiaryStoreTests.swift:453]:
```suggestion
		guard let eighteenDaysAgo = Calendar.current.date(byAdding: .day, value: -18, to: today) else {
```

File: src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift [hunk 4]
```diff
@@ -473,7 +479,10 @@ extension RiskProvider {
 				numberOfDaysWithLowRisk: risk.details.numberOfDaysWithRiskLevel,
 				numberOfDaysWithHighRisk: 0,
 				calculationDate: Date(),
-				riskLevelPerDate: [Date(): .low]
+				riskLevelPerDate: [
+					today: .low,
+					someDaysAgo: .high
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift:484]: If this mock should represent a low risk level, then all riskLevelsPerDate have to be .low
```suggestion
					someDaysAgo: .low
```

**Review by nickguendling — CHANGES_REQUESTED** [review]

**Review by nickguendling — APPROVED** [review]

**Review by pascal-brause — APPROVED** [review]

**Pull request merged**